### PR TITLE
ProcessManagers subscribe to rejections

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR adds the ability to subscribe to rejections from Process Managers.

Other notable changes:
 * Added `AbstractEntity.stringId()`
 * Improved returning IDs of event consuming entities.
 * `CommandHandlerMethod` now sets producer ID in `ThrowableMessage` for if `Entity` or `CommandHandler` thrown the message.

### CODE REVIEW

File: server/src/main/java/io/spine/server/procman/ProcessManager.java [hunk 1]
```diff
@@ -133,15 +136,28 @@ protected B getBuilder() {
...
+        checkNotNull(event);
 
+        final Message eventMessage = event.getMessage();
         final EventSubscriberMethod method = getMethod(getClass(), eventMessage);
-        method.invoke(this, eventMessage, context);
+        method.invoke(this, eventMessage, event.getEventContext());
+    }
+
+    /**
+     * Dispatched a rejection to the subscribing method of the process manager.
```
Comment (armiol) [inline server/src/main/java/io/spine/server/procman/ProcessManager.java]: `Dispatches`.

File: server/src/main/java/io/spine/server/route/RejectionRouting.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+import com.google.common.base.Optional;
+import com.google.protobuf.Message;
+import io.spine.core.RejectionClass;
+import io.spine.core.RejectionContext;
+
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
```
Comment (armiol) [inline server/src/main/java/io/spine/server/route/RejectionRouting.java]: Javadocs are missing.

**Comment by alexander-yevsyukov** [comment]
@armiol, now it's ready. PTAL.

File: server/src/test/java/io/spine/server/route/RejectionRoutingShould.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+/**
+ * @author Alexander Yevsyukov
+ */
+@SuppressWarnings("SerializableInnerClassWithNonSerializableOuterClass")
+// OK as custom routes do not refer to the test suite.
+public class RejectionRoutingShould {
+
+    /** The set of IDs returned by the {@link #defaultRoute}. */
+    private static final ImmutableSet<String> DEFAULT_ROUTE = ImmutableSet.of("α", "β", "γ");
```
Comment (armiol) [inline server/src/test/java/io/spine/server/route/RejectionRoutingShould.java]: καλή ιδέα!

**Pull request merged**